### PR TITLE
provider/google: Serialize saves infrastructure to bucket using Front 50, fixed project and access config fields

### PR DESCRIPTION
**Pull request by nwwebb**
Added a front 50 service for saving a snapshot of infrastructure to a gcs or aws bucket. Also added project fields to gce resources and fixed access configs. PTAL @lwander @duftler

### CODE REVIEW

File: clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/SerializeApplicationAtomicOperationUnitSpec.groovy [hunk 1]
```diff
@@ -170,7 +169,8 @@ class SerializeApplicationAtomicOperationUnitSpec extends Specification {
       def autoscalingMap = [name: SERVER_GROUP_NAME,
                             target: "\${google_compute_instance_group_manager.${SERVER_GROUP_NAME}.self_link}",
                             zone: SERVER_GROUP_ZONE,
-                            autoscaling_policy: autoscalingPolicyMap]
+                            autoscaling_policy: autoscalingPolicyMap,
+                            project: null]
```
Comment (lwander) [inline clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/SerializeApplicationAtomicOperationUnitSpec.groovy]: Are these necessary? Won't terraform just fill these in for us?

**Comment by lwander** [comment]
LGTM

File: clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/SerializeApplicationAtomicOperationUnitSpec.groovy (hunk 1 above)
Comment (nwwebb) [inline clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/SerializeApplicationAtomicOperationUnitSpec.groovy]: Filling out project isn't necessary for Terraform but serialization always occurs within the scope of an account so project should always get set by the atomic operation. For the tests I didn't set up an account, so the project is set to null by default. Do you think I should I mock an account for the tests or change the operation to not set project if an account is not specified (which should never actually happen)

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/SerializeApplicationAtomicOperation/SerializeApplicationAtomicOperation.groovy [hunk 2]
```diff
@@ -80,16 +84,14 @@ class SerializeApplicationAtomicOperation implements AtomicOperation<Void> {
   }
 
 
-  /* curl -X POST -H "Content-Type: application/json" -d '[ { "serializeApplication": { "applicationName": "codelab", "credentials": "my-google-account" }} ]' localhost:7002/gce/ops */
+  /* curl -X POST -H "Content-Type: application/json" -d '[ { "serializeApplication": { "applicationName": "exampleS", "credentials": "my-google-account" }} ]' localhost:7002/gce/ops */
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/SerializeApplicationAtomicOperation/SerializeApplicationAtomicOperation.groovy]: If the 'S' supposed to be capitalized in "exampleS"?

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/SerializeApplicationAtomicOperation/SerializeApplicationAtomicOperation.groovy [hunk 3]
```diff
@@ -290,10 +295,13 @@ class SerializeApplicationAtomicOperation implements AtomicOperation<Void> {
       if (networkInterface.subnetwork) {
         networkInterfaceMap.subnetwork = networkInterface.subnetwork.split("/").last()
       }
-      //TODO(nwwebb) deal with possible static ip here, empty map means ephemeral will be used
-      //TODO(nwwebb) check how the model differs when there is no external ip for an instance (empty list or null list)
-      if (networkInterface.accessConfigs instanceof List) {
+      if (networkInterface.accessConfigs != null) {
         networkInterfaceMap.access_configs = []
+        networkInterface.accessConfigs.each {AccessConfig accessConfig ->
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/SerializeApplicationAtomicOperation/SerializeApplicationAtomicOperation.groovy]: Missing a space between the `{` and `AccessConfig`.

**Comment by duftler** [comment]
LGTM after addressing comments.

**Pull request merged**